### DISCUSSION TITLE
SpaceDrive CLI Tool

### DISCUSSION THREAD

**Discussion by koalazub** [spacedriveapp/spacedrive#1229] [Feature Request]
Hey! I had a quick glance to see if this was mentioned, but haven't had such luck finding anything on the topic. 

I sometimes lean into using a VPS and don't have a GUI enabled. I can see this being immensely useful for some note taking apps and would like to know if there's been any talk of having a CLI for a frontend that we might be able to interact with as a daily driver? 

Use case: I work with an iPad, a VPS with no GUI, NixOS on a PC and a MacBook and require files to be sent between services using a GUI and not. 

Benefits: Having just a backend with the CLI attached to the front could be a good smoke detector to observe that things are going as planned. This could help with UI development as a means of being sure that invocations of all calls act as they should before you deal with UI as its own beast.

Foreseeable issues if implementing: Trying to auth via CLI could be problematic. So maybe following some of the other CLI tools for auth is an avenue that could be worth pursuing:

```nushell
# rough examples of what interacting with it could look like

# authorise for sync
space auth login

# options for logging in:
# - link to URL login
# - login via SSH
# - login via username/pwd

# sync nodes
space sync nodes

# sync directory
space sync
```

Having to instead sync things automatically using SpaceDrive vs `git` would be amazing as some projects are just small/ephemeral and I'm just looking to jot things down and look at them at another device down the track.

You could potentially follow other CLI tools such as [GitHub](https://cli.github.com/) or [rclone](https://rclone.org/#what) and have quick commands to invoke the same actions as the GUI tool you have. 

That being said, it would be a bit more work, so I can understand if it's not a consideration as yet(or ever).These are just spitball ideas and I hope that you found them to be worth considering/reading 

Either way, I'm excited as for this project to kick off!

**Comment by utkubakir** [comment]
Yup, we will have CLI support, it's under apps already and it's something we plan to support pretty soon (after release).

**Reply by koalazub** [reply]
Oh that's brilliant. Thanks for the response!
